woocommerce/woocommerce-gutenberg-products-block
# Deprecate legacy filter widgets with block alternatives

This PR gives us a method of soft-deprecating legacy filter widgets for Classic Themes in favour of Filter Blocks which were previously unable to be added to Classic Themes.


Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6144




### Screenshots

<img width="722" alt="Screenshot 2022-04-25 at 10 12 23" src="https://user-images.githubusercontent.com/8639742/165058480-bf2c49a7-8235-4a6c-90d7-45b2d7a54969.png">

### Testing

#### Automated Tests
### Manual Testing

How to test the changes in this Pull Request:

1. Checkout `trunk`.
2. Enable Storefront.
3. Go to Appearance > Widgets and add the following widgets to your sidebar: Filter by Attribute, Filter by Price, Active Filters.
4. Checkout this branch, and run `npm run build`.
5. Go to Appearance Widgets and on your legacy filter widgets, select it to show the toolbar, then select the far left icon.
6. Under the "Transform" dropdown you should be given an option to transform it to the block alternative filter. Do this for all your legacy filters.
7. Now add the Filter by Stock block. There was not a legacy version of this, but it's now available as a block.
8. Check that the legacy widgets you added in step 3 are no longer able to be added.
9. Check on the frontend that all these filters work as expected.


### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

1. With an active store using a Classic Theme such as Storefront that has the following legacy widgets in use: Filter by Attribute, Filter by Price, Active Filters.
2. Go to Appearance Widgets and on your legacy filter widgets, select it to show the toolbar, then select the far left icon.
3. Under the "Transform" dropdown you should be given an option to transform it to the block alternative filter. Do this for all your legacy filters.
4. Now add the Filter by Stock block. There was not a legacy version of this, but it's now available as a block.
5. Check that the legacy widgets you added in step 3 are no longer able to be added.
6. Check on the frontend that all these filters work as expected.

### Changelog

> Give merchants the option to convert their legacy filter widgets into filter blocks. Available for Filter by Attribute, Filter by Price, Active Filters and Filter by Stock.
